### PR TITLE
Various web app report improvements

### PR DESCRIPTION
**Pull request by tsteenbe**
Please see individual commits for details.

### CODE REVIEW

File: reporter-web-app/src/components/RuleViolationsTable.js [hunk 1]
```diff
@@ -156,6 +146,16 @@ const RuleViolationsTable = (props) => {
                         </span>
                     )
             ),
+            sorter: (a, b) => {
+                if (a.severityIndex < b.severityIndex) {
```
Comment (fviernau) [inline reporter-web-app/src/components/RuleViolationsTable.js:150]: Could we eliminate the code branches by just doing `return a.severityIndex - b.severityIndex` ?

File: reporter-web-app/src/App.css [hunk 2]
```diff
@@ -183,13 +183,17 @@ tr.ant-table-expanded-row, tr.ant-table-expanded-row:hover {
...
-.ort-table-operations button {
+/* Styling for 'Clear filters' and 'Shoe / Hide Columns' buttons in TableView */
+.ort-table-buttons .ant-btn-group {
     float: right;
-    margin: 16px 0px 16px 0px;
+    margin-bottom: 5px;
     z-index: 999;
 }
 
+.ort-table-toggle-columns .ant-dropdown-menu-item-selected {
```
Comment (fviernau) [inline reporter-web-app/src/App.css:193]: How about renaming `toggle-columns` to `package-filter` or `package-filter-menu` or `package-filter-selector`?

File: reporter-web-app/src/models/WebAppOrtResult.js [hunk 3]
```diff
@@ -393,6 +393,18 @@ class WebAppOrtResult {
...
     }
 
+    get repositoryConfiguration() {
+        return this.#repositoryConfiguration;
+    }
+
+    get ruleViolations() {
+        return this.#ruleViolations;
+    }
+
```
Comment (fviernau) [inline reporter-web-app/src/models/WebAppOrtResult.js:403]: I think we sometimes do this in the code base already.
But I always wonder why, e.g. wouldn't it make more sense to group closely related thing together?
I'm aware that the place in the alphabet can be a relation as well, but it usually starts to conflict with other (more important types of relations) at some point. I'm ok If you keep this commit but wanted to highlight the drawbacks at least.

File: reporter-web-app/config/webpack.config.js [hunk 4]
```diff
@@ -38,6 +38,8 @@ const shouldUseSourceMap = process.env.GENERATE_SOURCEMAP !== 'false';
 // makes for a smoother build process.
 const shouldInlineRuntimeChunk = process.env.INLINE_RUNTIME_CHUNK !== 'false';
 
+const isExtendingEslintConfig = process.env.EXTEND_ESLINT === 'true';
```
Comment (fviernau) [inline reporter-web-app/config/webpack.config.js:41]: commit-msg: In addition to updating the dev stack (dependencies) this commit applies a bunch of formatting fixes. Consider doing them in a separate commit.

File: reporter-web-app/src/App.css (hunk 2 above)
Comment (tsteenbe) [reply reporter-web-app/src/App.css:193]: A `filter` is Ant Design changes the number of row shown, whilst this option change the number of column visible.

File: reporter-web-app/config/webpack.config.js (hunk 4 above)
Comment (tsteenbe) [reply reporter-web-app/config/webpack.config.js:41]: The format fixes come from upstream not from me..

File: reporter-web-app/src/models/WebAppOrtResult.js (hunk 3 above)
Comment (tsteenbe) [reply reporter-web-app/src/models/WebAppOrtResult.js:403]: I am sorting alphabetically as that how it shown to me in it in my browser console / dev-utils and found out that organizing the code in this way helps me during development.

File: reporter-web-app/config/webpack.config.js (hunk 4 above)
Comment (fviernau) [reply reporter-web-app/config/webpack.config.js:41]: oh ok.

File: reporter-web-app/src/components/RuleViolationsTable.js (hunk 1 above)
Comment (tsteenbe) [reply reporter-web-app/src/components/RuleViolationsTable.js]: Added a separate commit to make this a one liner.

File: reporter-web-app/src/components/TableView.js [hunk 5]
```diff
@@ -20,17 +20,23 @@
 import React from 'react';
 import { connect } from 'react-redux';
 import {
+    Button,
```
Comment (sschuberth) [inline reporter-web-app/src/components/TableView.js:23]: Commit message: Why only to "several columns" and not all? How were the columns that got search added chosen? Please explain in the commit message.

Comment (tsteenbe) [reply reporter-web-app/src/components/TableView.js:23]: Updated the commit message - you can't have a column with search and drop-down so only added it to those without a drop-down

**Comment by fviernau** [comment]
@tsteenbe I just approved, maybe let's wait with the merge for @sschuberth as his last comment is still unresolved.

**Pull request merged**